api/src/main/java/uk/gov/dhsc/htbhf/claimant/entitlement/CycleEntitlementCalculator.java

```diff
@@ -20,38 +19,82 @@
     private final Integer entitlementCalculationDuration;
     private final Integer numberOfCalculationPeriods;
     private final EntitlementCalculator entitlementCalculator;
+    private final Integer weeksBeforeDueDate;
+    private final Integer weeksAfterDueDate;
 
-    public CycleEntitlementCalculator(@Value("${payment-cycle.cycle-duration-in-days}") Integer paymentCycleDurationInDays,
-                                      @Value("${payment-cycle.number-of-calculation-periods}") Integer numberOfCalculationPeriods,
+    /**
+     * Constructor for {@link CycleEntitlementCalculator}.
+     * @param paymentCycleConfig configuration for the payment cycle
+     * @param entitlementCalculator calculates entitlement for a single calculation period
+     */
+    public CycleEntitlementCalculator(PaymentCycleConfig paymentCycleConfig,
                                       EntitlementCalculator entitlementCalculator) {
-        validateArguments(paymentCycleDurationInDays, numberOfCalculationPeriods);
-        this.entitlementCalculationDuration = paymentCycleDurationInDays / numberOfCalculationPeriods;
-        this.numberOfCalculationPeriods = numberOfCalculationPeriods;
+        this.weeksBeforeDueDate = paymentCycleConfig.getWeeksBeforeDueDate();
+        this.weeksAfterDueDate = paymentCycleConfig.getWeeksAfterDueDate();
+        this.entitlementCalculationDuration = paymentCycleConfig.getEntitlementCalculationDuration();
```

Comment:
Should we just store `paymentCycleConfig` instead of defining each individual integer? Then just call them when required.

Reply:
I think in the code referencing weeksBeforeDueDate looks nice that paymentCycleConfig.getWeeksBeforeDueDate().